backend-lib/src/main/java/cash/z/ecc/android/sdk/internal/model/JniAddressCheckResult.kt

```diff
@@ -0,0 +1,17 @@
+package cash.z.ecc.android.sdk.internal.model
+
+import androidx.annotation.Keep
+
+/**
+ * Serves as cross layer (Kotlin, Rust) communication class.
+ */
+@Keep
+sealed class JniAddressCheckResult {
```

Comment:
I considered doing this and then intentionally didn't for `checkSingleUseTaddr`, instead returning `String?` across the JNI. This extra type serves us no purpose except to make the Kotlin side more complex, especially for `updateTransparentAddressTransactions` where we already know the address being checked so we only need a boolean reporting whether or not anything was found.

Remove this class.

Reply:
I disagree with this suggestion. Just like it is useful to have newtype wrappers to prevent incorrect use of primitive types, it's useful to have bespoke types here instead of booleans or bare strings. In addition, using a bespoke type here makes it easy to nondisruptively add information to the returned value if needed in the future.

Reply:
We don't have any examples of fetching a singleton data object instance from JNI, but I think I found the right incantation (but can't test it myself with my current setup.)